lib/graphiti/scoping/filter_group.rb

```diff
@@ -0,0 +1,81 @@
+module Graphiti
+  # @api private
+  module Scoping::FilterGroup
```

Comment:
I know you're still in draft mode and the functionality looks awesome so far. Wanted to jump in an offer one comment on this though before you get too far with it. 

Seems from quickly reading through that the only dependency on the `Scoping::Filter` class that you are mixing into is the `resource` method. What do you think about instead making this a `Scoping::FilterGroupValidator` class which accepts the resource as a collaborator? This could still be called in the same location with `FilterGroupValidator.new(resource).raise_unless...`, but would probably be a bit easier to test behavior and have less indirection in the form of a mixin as well.

Reply:
@wadetandy that sounds good. This is the sort of feedback I was hoping for posting the draft.